### PR TITLE
release-notes.txt: add 2023.07 release notes

### PR DESCRIPTION
**Pull request by benpicco**
### Contribution description
This are the release notes for the 2023.07 Release. Please give it a look and make sure that your work in the last 3 months is present here.

### Testing procedure
- Read the document

### CODE REVIEW

File: release-notes.txt [hunk 1]
```diff
@@ -1,3 +1,651 @@
...
+* makefiles/arch/msp430.inc.mk: Fix compilation issues with GCC 12 (#19484)
+* native/syscalls: rename real_clock_gettime to clock_gettime (#19476)
+* pkg/cmsis: use unique package for CMSIS headers, DSP and NN modules (#18056)
+* pkg/tinydtls: allow build for AVR (#19346)
+
+Device Drivers
+--------------
+
++ drivers/mtd_default: add external declarations for mtd* (#19538)
++ drivers: add support for MTDs emulated in RAM (#19443)
```
Comment (gschorcht) [inline release-notes.txt:186]:
```suggestion
+ drivers/mtd: add support for MTDs emulated in RAM (#19443)
+ drivers/mtd_default: add external declarations for mtd* (#19538)
```

**Review by gschorcht — CHANGES_REQUESTED** [review]

File: release-notes.txt [hunk 2]
```diff
@@ -1,3 +1,651 @@
...
+* dist/testbed-support: remove obsolete boards from iotlab archi (#19514)
+* dist/tools/insufficient_memory: fix collection of app folders (#19598)
+* dist/tools/openocd: fix parsing of flash bank base (#19620)
+* dist/tools/openocd: start debug-server in background and wait (#19737)
+* dist/tools/usb-serial: call ttys.py with its path [backport 2023.07] (#19831)
+* dist/tools: Use `/usr/bin/env` to lookup Python. (#19448)
+* make/usb_board_reset: define {preflash,term}-delay when necessary (#19617)
+* Makefile.include: don't use target lstfile implicitly (#19797)
+* Makefile.include: Generate lst file using objdump (#19745)
+* tests/drivers: move all driver tests into own folder (#19435)
```
Comment (aabadie) [inline release-notes.txt:217]: All `tests:` below should move to the Testing section.

File: release-notes.txt [hunk 3]
```diff
@@ -1,3 +1,651 @@
...
+* boards/nucleo-f429zi: Provide 2nd timer (#19447)
+* boards/rpi-pico: specify needed args when using jlink flasher (#19537)
+* boards/rpi-pico: update openocd.cfg file (#19717)
+* cpu/msp430: reorganize code (#19733)
+* cpu/nrf5x/periph/wdt: enable support for nRF53/nRF9160 (#19789)
+* cpu/nrf5x_common: rework LFCLK source selection (#19803)
+* cpu/rpx0xx: implement periph_spi (#19440)
+* cpu/rpx0xx: initial PIO support (#17425)
+* makefiles/tools/jlink.inc.mk: use ELF file for flashing (#19541)
+* tests: move all periph applications to their own periphs/ folder (#19552)
```
Comment (aabadie) [inline release-notes.txt:143]: All `tests:` and `treewide:` entries below should move to the Testing section

File: release-notes.txt [hunk 4]
```diff
@@ -1,3 +1,651 @@
...
+* tests/pkg/lvgl: avoid using floats (#19698)
+* tests/unittests: remove old workaround for SAML1X and gcc9.X bug (#19687)
+* tests: move leftover sys related tests to test/sys + move
+  tests/sys/candev to tests/drivers (#19585)
+* tests: move remaining driver related applications to tests/drivers (#19568)
+
+
+Uncategorized
+-------------
++ release-notes.txt: add 2023.04 release notes (#19503)
```
Comment (aabadie) [inline release-notes.txt:268]: I think this should be dropped.

File: release-notes.txt [hunk 5]
```diff
@@ -1,3 +1,651 @@
...
+* tests/lua_loader: move to tests/pkg/lua_loader (#19569)
+* tests/periph_rtt: Fix for tick conversion test (#19522)
+* tests/pkg/lvgl: avoid using floats (#19698)
+* tests/unittests: remove old workaround for SAML1X and gcc9.X bug (#19687)
+* tests: move leftover sys related tests to test/sys + move
+  tests/sys/candev to tests/drivers (#19585)
+* tests: move remaining driver related applications to tests/drivers (#19568)
+
+
+Uncategorized
```
Comment (aabadie) [inline release-notes.txt:266]: If there are uncategorized itmes, it means some labels were missing. This section should not show up in the notes.

File: release-notes.txt [hunk 6]
```diff
@@ -1,3 +1,651 @@
...
++ release-notes.txt: add 2023.04 release notes (#19503)
++ SUBSYSTEMD.md: add @kaspar030 to core, Rust, build system, CI &
+  timers (#19727)
++ SUBSYSTEMS.md: add @maribu (#19732)
++ SUBSYSTEMS.md: add jia200x to subsystems (#19512)
+* README.md: Remove HiL Badge (#19453)
+
+And 65 minor changes.
+
+Deprecations
```
Comment (aabadie) [inline release-notes.txt:277]: No deprecation, so just remove (something is missing in the script it seems)

File: release-notes.txt [hunk 7]
```diff
@@ -1,3 +1,651 @@
...
+* EXTERNAL_MODULE_DIRS silently ignores non-existent entries (#17696)
+* make: ccache leads to differing binaries (#14264)
+* make: use of immediate value of variables before they have their
+  final value (#8913)
+* missing build dependencies in the rust build (#19714)
+
+Other issues (64)
+-----------------
+
+* 2023.04 release bug tracking (#19469)
```
Comment (aabadie) [inline release-notes.txt:501]: This should not show up in the notes

**Review by aabadie — COMMENTED** [review]
Some initial comments in the generated list of PRs/issues

**Review by gschorcht — APPROVED** [review]
Looks good from my point of view, so 1 ACK from my side. Needs squashing of course.

File: release-notes.txt [hunk 8]
```diff
@@ -1,3 +1,641 @@
...
++ cpu/rpx0xx: add initial ADC support (#19516)
++ drivers/usbdev_synopsys_dwc2: add GD32V support (#19389)
+* boards/adafruit-clue: use shared usb_board_reset.mk for
+  flash/reset/term targets (#19616)
+* boards/arduino-mkrwan1300: update board name to correct one (#19526)
+* boards/atmega8: new board (#19754)
+* boards/esp32: deduplication in common ESP32x board definitions (#19432)
+* boards/frdm-k22f: fixes long lines in periph_conf.h (#19774)
+* boards/frdm-k64f: fixes long lines and comma separated by
+  whitespaces… (#19775)
```
Comment (MrKevinWeiss) [inline release-notes.txt:132]:
```suggestion
```

File: release-notes.txt [hunk 9]
```diff
@@ -1,3 +1,641 @@
...
+* boards/nucleo-f429zi: Provide 2nd timer (#19447)
+* boards/rpi-pico: specify needed args when using jlink flasher (#19537)
+* boards/rpi-pico: update openocd.cfg file (#19717)
+* cpu/msp430: reorganize code (#19733)
+* cpu/nrf5x/periph/wdt: enable support for nRF53/nRF9160 (#19789)
+* cpu/nrf5x_common: rework LFCLK source selection (#19803)
+* cpu/rpx0xx: implement periph_spi (#19440)
+* cpu/rpx0xx: initial PIO support (#17425)
+* makefiles/tools/jlink.inc.mk: use ELF file for flashing (#19541)
+* treewide: fix typos and false positives found by codespell (#19528)
```
Comment (MrKevinWeiss) [inline release-notes.txt:143]:
```suggestion
```

**Review by MrKevinWeiss — COMMENTED** [review]
Just the few small things then I would ack.

**Comment by benpicco** [comment]
bors merge

**Comment by benpicco** [comment]
Thank you!

**Pull request merged**